plugins/check_perfmon.cpp

```diff
@@ -314,27 +304,26 @@ BOOL QueryPerfData(printInfoStruct& pI)
 		goto die;
 
 	/*
-	 * Most counters need two queries to provide a value.
-	 * Those which need only one will return the second.
-	 */
+	* Most counters need two queries to provide a value.
+	* Those which need only one will return the second.
+	*/
 	Sleep(pI.dwPerformanceWait);
 
 	status = PdhCollectQueryData(hQuery);
 	if (FAILED(status))
 		goto die;
 
-	status = PdhGetFormattedCounterArray(hCounter, pI.dwRequestedType, &dwBufferSize, &dwItemCount, pDisplayValues);
+	status = PdhGetFormattedCounterArray(hCounter, pI.dwRequestedType, &dwBufferSize, &dwItemCount, NULL);
 	if (status != PDH_MORE_DATA)
 		goto die;
 
-	pDisplayValues = reinterpret_cast<PDH_FMT_COUNTERVALUE_ITEM*>(new BYTE[dwBufferSize]);
+	PDH_FMT_COUNTERVALUE_ITEM *pDisplayValues = reinterpret_cast<PDH_FMT_COUNTERVALUE_ITEM*>(new BYTE[dwBufferSize]);
```

Comment:
pDisplayValues needs to be declared before we can run into a `goto die` as it is referenced there.